modules/azure/scalesets.go

```diff
@@ -0,0 +1,153 @@
+package azure
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2019-07-01/compute"
+	"github.com/gruntwork-io/terratest/modules/testing"
+	"github.com/stretchr/testify/require"
+)
+
+// GetVirtualMachineScaleSetsClient is a helper function that will setup an Azure Virtual Machine Scale Sets client on your behalf.
+// This function would fail the test if there is an error.
+func GetVirtualMachineScaleSetsClient(t testing.TestingT, subscriptionID string) *compute.VirtualMachineScaleSetsClient {
+	vmssClient, err := GetVirtualMachineScaleSetsClientE(subscriptionID)
+	require.NoError(t, err)
+	return vmssClient
+}
+
+// GetVirtualMachineScaleSetsClientE is a helper function that will setup an Azure Virtual Machine Scale Sets client on your behalf.
+func GetVirtualMachineScaleSetsClientE(subscriptionID string) (*compute.VirtualMachineScaleSetsClient, error) {
+	// Create a VM Scale Sets client
+	vmssClient, err := CreateVirtualMachineScaleSetsClientE(subscriptionID)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create an authorizer
+	authorizer, err := NewAuthorizer()
+	if err != nil {
+		return nil, err
+	}
+
+	// Attach authorizer to the client
+	vmssClient.Authorizer = *authorizer
+	return vmssClient, nil
+}
+
+// VirtualMachineScaleSetExists indicates whether the specifcied Azure Virtual Machine Scale Set exists.
+// This function would fail the test if there is an error.
+func VirtualMachineScaleSetExists(t testing.TestingT, vmssName string, resGroupName string, subscriptionID string) bool {
+	exists, err := VirtualMachineScaleSetExistsE(vmssName, resGroupName, subscriptionID)
+	require.NoError(t, err)
+	return exists
+}
+
+// VirtualMachineScaleSetsExistsE indicates whether the specifcied Azure Virtual Machine exists.
+func VirtualMachineScaleSetExistsE(vmssName string, resGroupName string, subscriptionID string) (bool, error) {
+	// Get VM Object
+	_, err := GetVirtualMachineScaleSetE(vmssName, resGroupName, subscriptionID)
+	if err != nil {
+		if ResourceNotFoundErrorExists(err) {
+			return false, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
+// GetVirtualMachineScaleSetCapacity gets the capacity of a Virtual Machine Scale Set.
```

Comment:
please keep only the generic helpers, then use the test to show how to extract the VirtualMachineScaleSet object details. similar to this test [terratest/test/azure/terraform_azure_functionapp_example_test.go](https://github.com/gruntwork-io/terratest/blob/23b214b8a30ff7a371244ecb6ea626b44e347592/test/azure/terraform_azure_functionapp_example_test.go#L47)